### ISSUE TITLE
variable url parameter

### ISSUE THREAD

**Issue by pokryva** [pokiiio/node-red-contrib-web-page-screenshot#4]
Could you please modify the node to take "Web Page URL" as an incoming parameter? For instance msg.url, or msg.payload.url?

**Comment by pokiiio** [comment]
@pokryva 
Thank you for your bright idea. Let me code it.

**Comment by kumajaya** [comment]
For now you must access puppeteer directly, check https://nodered.org/docs/user-guide/writing-functions#loading-additional-modules. Add puppeteer to settings.js file, restart Node-RED:
```
functionGlobalContext: {
    puppeteer:require('puppeteer')
}
```
Write in your Function node:
```
let url = msg.url;
let path = null;
let selector = null;
//let selector = 'body > div.ms-site-container > div.container > div > div > div';
let puppeteer = global.get('puppeteer');
let option = {};

if (!url) {
	// set to default.
	url = 'http://www.example.com/';
}

if (path) {
	option.executablePath = path;
}

puppeteer.launch(option).then(async browser => {
	let option = {};

	const page = await browser.newPage();
	await page.setViewport({ width: 1280, height: 600 });
	await page.goto(url);
	// disable animation
	//await page.evaluate(() => document.querySelector('body > div.ms-site-container > div.container > div > div > div').setAttribute('style', 'animation: none'));

	if (selector) {
		const clip = await page.$eval(selector, item => {
			const {
				width,
				height,
				top: y,
				left: x
			} = item.getBoundingClientRect()
			return {
				width,
				height,
				x,
				y
			}
		});

		option = {
			clip,
			type: 'png',
			fullPage: false,
			encoding: 'base64'
		};
	} else {
		option = {
			type: 'png',
			fullPage: true,
			encoding: 'base64'
		};
	}

	await page.waitFor(1000);
	const base64String = await page.screenshot(option);
	await browser.close();

	msg.payload = base64String;
	node.send(msg);
});
```
I hope this node more flexible in the future.

**Comment by pokiiio** [comment]
thanks for code, @kumajaya !

**Comment by pokiiio** [comment]
@pokryva 
this feature is now available on v0.2.1.
https://github.com/pokiiio/node-red-contrib-web-page-screenshot/releases/tag/v0.2.1
